### PR TITLE
#34  error message hyperlink with jobid

### LINKED ISSUE #34: [BUG] Catch error message when Retrieve function failes
**Current Behavior**
  
When an error occurs while Refreshing metadata in commit page, the user sees no valuable information to resolve the issue.
  
image (1)

  
**Expected Behavior**
  
Catch error message returned by Function execution method in apex.
  
  
**Steps To Reproduce**

@DannyCastroVentura  + @anil-sfdc :
1. make retrieve function `throw new Error()` (up top in `run()`) @DannyCastroVentura 
3. Click on Refresh button @anil-sfdc 
4. fix retrieve function again @DannyCastroVentura 
5. reload page @anil-sfdc 
6. make commit function `throw new Error()` (up top in `run()`) @DannyCastroVentura 
7. Click on Commit Changes @anil-sfdc 
8. fix commit function again @DannyCastroVentura 

ideal outcome:
tell the user an error happened and that more details can be found on the related RESULT or JobExectution record that was created for the refresh/commit action

### PR DESCRIPTION
**Pull request by anil-sfdc**
# PR details

## What is the purpose of this pull request? (put an "X" next to an item)

- [ ] Bug fix
- Error message with job execution  Id. user click and navigate the error detail page. 
- [ ] Add a GUI option
- Error message with job execution Id. user click and navigate the error detail page. 

## What changes did you make? (Give an overview)

closes #34

### CODE REVIEW

File: force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js [hunk 1]
```diff
@@ -351,10 +351,25 @@ export default class MarketingCloudCopadoDataTable extends LightningElement {
...
         } else if (jobExecution.copado__Status__c === "Error") {
             this.loadingState(false);
-            this.showError(
-                ` An error occurred while Metadata file Retrieve.please refere to the job id: `,
-                jobExecutionId
-            );
+            let JobUrl =
+                "https://copadonull71743980.lightning.force.com/lightning/r/copado__JobExecution__c/" +
+                jobExecutionId +
+                "/view";
```
Comment (JoernBerkefeld) [inline force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js:357]:
```suggestion
            let JobUrl = "/" +jobExecutionId;
```

File: force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js [hunk 2]
```diff
@@ -351,10 +351,25 @@ export default class MarketingCloudCopadoDataTable extends LightningElement {
...
+            const errEvent = new ShowToastEvent({
+                title: "Error",
+                variant: "error",
+                mode: "sticky",
+                message: "Refreshing metadata list failed. For more details {0} please go to {1}",
+                messageData: [
+                    "Salesforce",
```
Comment (JoernBerkefeld) [inline force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js:364]:
```suggestion
```

File: force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js [hunk 3]
```diff
@@ -351,10 +351,25 @@ export default class MarketingCloudCopadoDataTable extends LightningElement {
...
-            );
+            let JobUrl =
+                "https://copadonull71743980.lightning.force.com/lightning/r/copado__JobExecution__c/" +
+                jobExecutionId +
+                "/view";
+            const errEvent = new ShowToastEvent({
+                title: "Error",
+                variant: "error",
+                mode: "sticky",
+                message: "Refreshing metadata list failed. For more details {0} please go to {1}",
```
Comment (JoernBerkefeld) [inline force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js:362]:
```suggestion
                message: "Refreshing metadata list failed. For more details {0}.",
```

File: force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js [hunk 4]
```diff
@@ -351,10 +351,25 @@ export default class MarketingCloudCopadoDataTable extends LightningElement {
...
+                messageData: [
+                    "Salesforce",
+                    {
+                        url: JobUrl,
+                        //  url: "https://copadonull71743980.lightning.force.com",
+                        label: "this Job Execution record"
```
Comment (JoernBerkefeld) [inline force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js:368]:
```suggestion
                        label: "click here"
```

**Review by JoernBerkefeld — CHANGES_REQUESTED** [review]

File: force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js [hunk 5]
```diff
@@ -351,10 +351,21 @@ export default class MarketingCloudCopadoDataTable extends LightningElement {
...
+            let JobUrl = "/" +jobExecutionId;
+            const errEvent = new ShowToastEvent({
+                title: "Error",
+                variant: "error",
+                mode: "sticky",
+                message: "Refreshing metadata list failed. For more details {0}.",
+                messageData: [
+                    {
+                        url: JobUrl,
+                        //  url: "https://copadonull71743980.lightning.force.com",
```
Comment (JoernBerkefeld) [inline force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js:363]:
```suggestion
```

**Comment by anil-sfdc** [comment]
image
 After update pram 1

File: force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js [hunk 6]
```diff
@@ -351,10 +351,20 @@ export default class MarketingCloudCopadoDataTable extends LightningElement {
...
-            this.showError(
-                ` An error occurred while Metadata file Retrieve.please refere to the job id: `,
-                jobExecutionId
-            );
+            let JobUrl = "/" + jobExecutionId;
+            const errEvent = new ShowToastEvent({
+                title: "Error",
+                variant: "error",
+                mode: "sticky",
+                message: "Refreshing metadata list failed. For more details {0}.{1}",
```
Comment (JoernBerkefeld) [inline force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js:359]: why did you add the "{1}"?

**Review by JoernBerkefeld — CHANGES_REQUESTED** [review]

**Review by JoernBerkefeld — APPROVED** [review]

**Pull request merged**